Moving Sarifer code from SARIF VS Viewer repositoy

yongyan-gh opened the pull request:
# Description
- There is no source code change compared to original files.
- Updated the build related scripts to build only Sarifer extensions.
- Removed the submodule references to Sarif-PatternMatcher repo, use nuget package instead.
- Added the submodule to Sarif-VisualStudio-Extension to reuse shared libraies.

EasyRhinoMSFT commented on .github/CODEOWNERS:
```diff
@@ -0,0 +1,5 @@
+# This file controls who is tagged for review for any given pull request.
+
+# For anything not explicitly taken by someone else:
+* @EasyRhinoMSFT @michaelcfanning @marmegh
```
>@marmegh [](http://example.com/codeflow?start=34&length=8)

Remove Mary
Add hulonj and suvamM

EasyRhinoMSFT replied:
^ these are their GH usernames

EasyRhinoMSFT commented:
I see a file called "sarif-visualstudio-extension" in the src folder, what is it?

yongyan-gh commented:
Description
- There is no source code change compared to original files.
- Updated the build related scripts to build only Sarifer extensions.
- Removed the submodule references to Sarif-PatternMatcher repo, use nuget package instead.
- Added the submodule to Sarif-VisualStudio-Extension to reuse shared libraies.

yongyan-gh commented:
Its the sarif-visualstudio-extension submodule. Sarifer references to the Sarif.Viewer.VisualStudio.Interop project and maybe the shared Sarif.Viewer.VisualStudio.Shell/2022 projects in the future.

---
In reply to: [1278289434](https://github.com/microsoft/sarifer/pull/2#issuecomment-1278289434) [](http://example.com/codeflow?ancestors=1278289434,1278289434)

EasyRhinoMSFT commented:
Ah, right

---
In reply to: [1279251021](https://github.com/microsoft/sarifer/pull/2#issuecomment-1279251021) [](http://example.com/codeflow?ancestors=1279251021,1278289434)

EasyRhinoMSFT commented on .github/CODEOWNERS:
```diff
@@ -0,0 +1,5 @@
+# This file controls who is tagged for review for any given pull request.
+
+# For anything not explicitly taken by someone else:
+* @EasyRhinoMSFT @michaelcfanning
```
Please add @hulonj and @suvamM

yongyan-gh replied:
updated

EasyRhinoMSFT commented on scripts/New-AtomXml.ps1:
```diff
@@ -0,0 +1,86 @@
+        <author>
+            <name>1ES Security Tools Team</name>
+        </author>
+        <content type="application/octet-stream" src="$FeedUri-$VsVersion/Microsoft.Sarif.Sarifer.vsix"/>
```
>$FeedUri-$VsVersion [](http://example.com/codeflow?start=54&length=19)

Change to:

$FeedUri-sarifer-$VsVersion

yongyan-gh replied:
updated

EasyRhinoMSFT commented on scripts/New-AtomXml.ps1:
```diff
@@ -0,0 +1,86 @@
+            <DownloadCount xsi:nil="true"/>
+        </Vsix>
+        <link rel="releasenotes" type="text/markdown" href="https://github.com/microsoft/sarif-visualstudio-extension/blob/master/src/ReleaseHistory.md"/>
+        <link rel="icon" href="$FeedUri-$VsVersion/Triskele.ico"/>
```
>$FeedUri-$VsVersion/Triskele.ico [](http://example.com/codeflow?start=31&length=32)

Change to

$FeedUri-$VsVersion/Resources/Package.ico

yongyan-gh replied:
updated

EasyRhinoMSFT commented on src/Sarif.Sarifer.Core/SariferPackage.cs:
```diff
@@ -0,0 +1,129 @@
+    public sealed class SariferPackage : AsyncPackage, IDisposable
+    {
+        public const string PackageGuidString = "F70132AB-4095-477F-AAD2-81D3D581113B";
+        public const string OptionCategoryName = "Sarifer";
```
>Sarifer [](http://example.com/codeflow?start=50&length=7)

I would like to start spelling this "SARIFer". SARIF is an abbreviation so it should always be all-caps.  And the 'er' is not part of the abbreviation, so it should be lowercase.

EasyRhinoMSFT replied:
I know it's a pain, but let's do a pass for all strings that are user-facing. Don't worry about code comments, etc.

yongyan-gh replied:
Theses need to be const because they are used in the attribute. Moved them to Constants.cs.

Also added the resources.resx for other user faced strings e.g. option page names, output pane logs.

EasyRhinoMSFT reviewed (approved):
:shipit:
